extension/app/ts/utils/bigint.ts

```diff
@@ -140,7 +140,7 @@ export function squareRoot(value: bigint) {
 
 export function stringifyJSONWithBigInts(value: any): string {
 	return JSON.stringify(value, (_key, value) => {
-		return typeof value === "bigint" ? `0x${value.toString(16)}n` : value
+		return typeof value === "bigint" ? `0x${ value.toString(16) }` : value
```

Comment:
The `n` here makes it so you can automatically identify which hex strings are bigints and which are binary data.  It allows us to put something like this into a JSON.parse decoder:
```ts
if (typeof x === 'string') {
	if (/^0x[a-fA-F0-9]+n$/.test(x)) return BigInt(x)
	else if (/^0x[a-fA-F0-9]+$/.test(x)) return hexStringToUint8Array(x)
} else {
	decodeAsNormalJson(x)
}
```

Reply:
the problem with that identification is that if you write a string that is not intended to be bigint, it just becomes bigint. Like for example (not actually done) you have a token name "0xn" would become come bigint 0, while its supposed to be a name of a token.

Reply:
While I agree it is possible to have a string like `0x0n` that was meant to just be that string, it seems *incredibly* unlikely and not worth the cost of choosing to have byte arrays and numbers encoded the same way (thus cannot be auto-decoded).